lib/fluent/plugin/out_google_cloud_storage_out.rb

```diff
@@ -136,37 +139,52 @@ def chunk_unique_id_to_str(unique_id)
       unique_id.unpack('C*').map{|x| x.to_s(16).rjust(2,'0')}.join('')
     end
 
-    def path_format_with_unique_strategy(path, strategy, chunk_key, chunk_unique)
-       case strategy
-       when nil
-         path
-       when :chunk_id
-         path.gsub(UNIQUE_PLACE_HOLDER, chunk_unique_id_to_str(chunk_unique))
-       when :increment
-         if @before_chunk_key
-           if @before_chunk_key == chunk_key
-             @samepath_counter += 1
-           else
-             @samepath_counter = 0
-           end
-         end
-         @before_chunk_key = chunk_key
-         path.gsub(UNIQUE_PLACE_HOLDER, "#{@samepath_counter}")
-       when :timestamp
-         path.gsub(UNIQUE_PLACE_HOLDER, Time.now.strftime(@unique_format))
-       end
+    def path_format_with_unique_strategy(path, strategy, chunk_key, chunk_unique, tag)
+      if tag
+        path = path.gsub('${tag}', tag)
+        hash = {}
+        tag.split('.').each_with_index do |part, i|
```

Comment:
We can use `#extract_placeholder(@name_of_parameter, chunk.metadata)` instead of hand written method.
Could you use this API?